tests/route-spec/middleware.test.ts

```diff
@@ -1,254 +1,248 @@
+    },
+    routeSpec: {
+      auth: "none",
+      methods: ["GET"],
+    },
+    routeFn: (req) => {
+      t.is(req.name, "lucille")
+      return new Response()
+    },
+    routePath: "/hello",
   })
 
-  await withEdgeSpec({
-    auth: "none",
-    methods: ["GET"],
-  })((req) => {
-    t.is(req.name, "lucille")
-    return new Response()
-  })({} as EdgeSpecRequest)
+  await axios.get("/hello")
 })
 
 test("receives local middleware", async (t) => {
-  const withEdgeSpec = createWithEdgeSpec({
-    apiName: "hello-world",
-    productionServerUrl: "https://example.com",
+  const { axios } = await getTestRoute(t, {
+    globalSpec: {
+      apiName: "hello-world",
+      productionServerUrl: "https://example.com",
 
-    authMiddlewareMap: {},
-    globalMiddlewares: [],
+      authMiddlewareMap: {},
+      globalMiddlewares: [],
+    },
+    routeSpec: {
+      auth: "none",
+      methods: ["GET"],
+      middlewares: [withName],
+    },
+    routeFn: (req) => {
+      t.is(req.name, "lucille")
+      return new Response()
+    },
+    routePath: "/hello",
   })
 
-  await withEdgeSpec({
-    auth: "none",
-    methods: ["GET"],
-    middlewares: [withName],
-  })((req) => {
-    t.is(req.name, "lucille")
-    return new Response()
-  })({} as EdgeSpecRequest)
+  await axios.get("/hello")
 })
 
 test("responseDefaults are passed", async (t) => {
-  const withEdgeSpec = createWithEdgeSpec({
-    apiName: "hello-world",
-    productionServerUrl: "https://example.com",
-
-    authMiddlewareMap: {},
-    globalMiddlewares: [
-      () => ({
-        responseDefaults: new Response(null, {
-          headers: new Headers({ "x-test": "test", "x-test2": "test2" }),
-        }),
-      }),
-    ],
-  })
-
-  const response = await withEdgeSpec({
-    auth: "none",
-    methods: ["GET"],
-    middlewares: [
-      () => ({
-        responseDefaults: new Response("body text", {
-          headers: new Headers({ "x-test": "test2" }),
-        }),
-      }),
-    ],
-  })(() => {
-    return new Response()
-  })({} as EdgeSpecRequest)
-
-  t.is(await streamToString(response.body), "body text")
-  t.is(response.headers.get("x-test"), "test2")
-  t.is(response.headers.get("x-test2"), "test2")
-})
-
-test("responseDefaults are passed (responseDefaults as ResponseInit)", async (t) => {
-  const withEdgeSpec = createWithEdgeSpec({
-    apiName: "hello-world",
-    productionServerUrl: "https://example.com",
-
-    authMiddlewareMap: {},
-    globalMiddlewares: [
-      () => ({
-        responseDefaults: {
-          headers: { "x-test": "test", "x-test2": "test2" },
+  const { axios } = await getTestRoute(t, {
+    globalSpec: {
+      apiName: "hello-world",
+      productionServerUrl: "https://example.com",
+
+      authMiddlewareMap: {},
+      globalMiddlewares: [
+        (next, req) => {
+          req.responseDefaults.headers.set("x-test", "test")
+          req.responseDefaults.headers.set("x-test2", "test2")
```

Comment:
🐧 I don't want to cause more thrashing but this smells a little funny, what do you think about this vs. `ctx.req`, `ctx.res`, and `ctx.env.custom_application_property`? kinda how hono does it:

https://hono.dev/api/context#context
https://hono.dev/concepts/middleware

Reply:
Seve seemed pretty insistent on keeping compatibility with nextlove on this, and moving as quickly as possible and not worrying about api design, so idk, maybe we can worry about it later?

Reply:
I mean the current design isn't compatible with nextlove either haha

but yeah ok going with this for now and maybe changing the interface later

Reply:
> I mean the current design isn't compatible with nextlove either haha

I guess I meant, we wanna keep the `request` object at top-level for now